src/br/unicamp/meca/mind/MecaMind.java

```diff
@@ -277,140 +293,189 @@ private void mountMotivationalCodelets() {
-								if (motorCodelet.getId()
-										.equalsIgnoreCase(motivationalBehavioralCodelet.getMotorCodeletId())) {
-									motivationalBehavioralCodelet.addOutputs(motorCodelet.getInputs());
-								}
-							}
-						}
-					}
+					 */										
+					behaviorCodelet.addOutput(actionSequencePlanMemoryContainer);
+					behaviorCodelet.addOutput(actionSequencePlanRequestMemoryContainer);
+					
 					/*
 					 * Inputs
 					 */
-					if (getMotivationalCodelets() != null) {
-						for (MotivationalCodelet motivationalCodelet : getMotivationalCodelets()) {
+					if (motivationalCodelets != null) {
+						for (MotivationalCodelet motivationalCodelet : motivationalCodelets) {
 							if (motivationalCodelet != null && motivationalCodelet.getId() != null) {
-								ArrayList<String> motivationalCodeletsIds = motivationalBehavioralCodelet
+								ArrayList<String> motivationalCodeletsIds = behaviorCodelet
 										.getMotivationalCodeletsIds();
 								if (motivationalCodeletsIds != null) {
 									for (String motivationalCodeletId : motivationalCodeletsIds) {
 										if (motivationalCodeletId != null && motivationalCodelet.getId()
 												.equalsIgnoreCase(motivationalCodeletId)) {
-											motivationalBehavioralCodelet.addInputs(motivationalCodelet.getOutputs());
+											behaviorCodelet.addInputs(motivationalCodelet.getOutputs());
+										}
+									}
+								}
+							}
+						}
+					}
+					
+					if(perceptualCodelets != null) {
+						for(PerceptualCodelet perceptualCodelet : perceptualCodelets) {
+							if(perceptualCodelet != null && perceptualCodelet.getId() != null) {
+								ArrayList<String> perceptualCodeletsIds = behaviorCodelet.getPerceptualCodeletsIds();
+								if(perceptualCodeletsIds != null) {
+									for(String perceptualCodeletId : perceptualCodeletsIds) {
+										if(perceptualCodeletId != null && perceptualCodelet.getId().equalsIgnoreCase(perceptualCodeletId)) {
+											behaviorCodelet.addInputs(perceptualCodelet.getOutputs());
 										}
 									}
 								}
 							}
 						}
 					}
 
-					if (soarCodelet != null && soarCodelet.getId() != null && motivationalBehavioralCodelet.getSoarCodeletId() != null) {
-						if (soarCodelet.getId().equalsIgnoreCase(motivationalBehavioralCodelet.getSoarCodeletId())) {
-							motivationalBehavioralCodelet.addBroadcasts(soarCodelet.getOutputs());
+					if (soarCodelet != null && soarCodelet.getId() != null && behaviorCodelet.getSoarCodeletId() != null) {
+						if (soarCodelet.getId().equalsIgnoreCase(behaviorCodelet.getSoarCodeletId())) {
+							behaviorCodelet.addBroadcasts(soarCodelet.getOutputs());
 						}
 
 					}
 
-					insertCodelet(motivationalBehavioralCodelet);
+					insertCodelet(behaviorCodelet);
 
 				}
 			}
 		}
 	}
+	
+	private void mountActionFromPerceptionCodelets() {
+		if (actionFromPerceptionCodelets != null) {
+			for (ActionFromPerception actionCodelet : actionFromPerceptionCodelets) {
```

Comment:
actionFromPerceptionCodelets.stream().forEach(........

Reply:
Nice suggestion, indeed, but as is just a matter of style, resulting in the same end, will let it to a future PR.